lib/payslip/payslip.rb

```diff
@@ -0,0 +1,19 @@
+require_relative '../data_model/tax_bracket'
+
+module Payslip
+  def generate_payslip()
+    raise "Not implemented"
+  end
+
+  private def calculate_gross_income
+    raise "Not implemented"
+  end
+
+  private def calculate_income_tax
+    raise "Not implemented"
+  end
+
+  private def calculate_net_income
+    raise "Not implemented"
+  end
+end
```

Comment:
in ruby there is no concept of the interface because of its nature being dynamic
the way we enforce it is by unit test something like this

expect(obj.responds_to?(:function))